app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -502,6 +506,8 @@ protected void onPause() {
         if (adapter != null) {
             adapter.unregisterDataSetObserver(mListStateObserver);
         }
+
+        hereFunctionHandler.stopLocationUpdates();

```

Comment:
One tricky thing about stopping in `onPause` is that you may want to resume later and continue acquiring a location. For instance, if you are getting a location and switch to another app for a sec and switch back, you will have paused and resumed this activity, but there is no restart code for the location acquisition logic. Doing this cleanly is sorta tricky, we can talk further about possible approaches.